src/sqs/index.ts

```diff
@@ -345,28 +359,50 @@ export class SqsQueue<MessagePayload> {
 			this.sendProcessedMetrics(messageProcessingDuration);
 			await this.deleteMessage(message, log)
 		} catch (err) {
+			await this.handleSqsMessageExecutionError(err, context, log, message);
+		}
+	}
 
-			statsd.increment(sqsQueueMetrics.failed, this.metricsTags)
-			log.error({err}, "Error while executing SQS message")
-
-			try {
-				const errorHandlingResult = await this.errorHandler(err, context);
-				if (!errorHandlingResult.retryable) {
-					log.info("Deleting the message because it is not retryable")
-					await this.deleteMessage(message, log)
-				} else if (errorHandlingResult.skipDlq && context.receiveCount >= this.maxAttempts) {
-					log.info("Deleting the message because it has reached the maximum amount of retries")
-					await this.deleteMessage(message, log)
-				} else if (errorHandlingResult.retryDelaySec !== undefined /*zero seconds delay is also supported*/) {
-					log.info(`Delaying the retry for ${errorHandlingResult.retryDelaySec} seconds`)
-					await this.changeVisabilityTimeout(message, errorHandlingResult.retryDelaySec, log);
-				}
-			} catch (err) {
-				log.error({err}, "Error while performing error handling");
+	private async handleSqsMessageExecutionError(err, context: Context<MessagePayload>, log: LoggerWithTarget, message: SQS.Message) {
+		try {
+			const errorHandlingResult = await this.errorHandler(err, context);
+
+			if (errorHandlingResult.isFailure) {
+				log.error({err}, "Error while executing SQS message")
+				statsd.increment(sqsQueueMetrics.failed, this.metricsTags)
+			} else {
+				log.info({err}, "Expected exception while executing SQS message. Not an error, deleting the message.")
 			}
+
+			if (isNotAFailure(errorHandlingResult)) {
+				log.info("Deleting the message because the error is not a failure")
+				await this.deleteMessage(message, log)
+			} else if (isNotRetryable(errorHandlingResult)) {
+				log.info("Deleting the message because the error is not retryable")
```

Comment:
Should it be a warning? That's a error scenario, right?...

Reply:
Technically, it is kind of expected error which we treat as a "success", but yea I can make it warn